### DISCUSSION TITLE
A Controlled MenuItemSwitch

### DISCUSSION THREAD

**Discussion by musyoka-muthangya** [microsoft/fluentui#31901] [Q&A] [answered]
It is impossible to use the MenuItemSwitch in a controlled way:

```
const [checked, setChecked] = useState(true);
const onChange = async () => {
  const success = await doSomeWork();

 // only toggle the switch after doSomeWork is finished
 setChecked(success);
};
<MenuItemSwitch
   name="nam"
   value="val
   checked={checked}">
        Switch
 </MenuItemSwitch>
```

**Comment by layershifter** [comment, marked as answer]
`MenuItemSwitch` have the same behavior as `MenuItemCheckbox` and cannot be controlled individually. They should be controlled at `Menu`/`MenuGroup` level, please check https://react.fluentui.dev/?path=/docs/components-menu-menu--default#controlled-checkbox-items
